sympy/assumptions/ask.py

```diff
@@ -363,6 +363,56 @@ def _extract_all_facts(assump, exprs):
                 facts.add(frozenset(args))
     return CNF(facts)
 
+def _normalize_predicate(expr):
```

Comment:
We shouldn't have to write a full recursive function here. We should be able to just write down a dictionary of replacements and use the `replace` method https://docs.sympy.org/latest/modules/core.html#sympy.core.basic.Basic.replace